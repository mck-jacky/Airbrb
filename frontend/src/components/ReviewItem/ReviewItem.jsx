import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import AccountCircleOutlinedIcon from '@mui/icons-material/AccountCircleOutlined';
import { Box, Rating, Typography } from '@mui/material';

const ReviewItem = ({ review, additionalStyle }) => {
  return (
    <Box data-testid='review-item'>
      <AccountCircleOutlinedIcon sx={{ fontSize: 44, position: 'relative', top: 8, left: -5 }} />
      <Box sx={{ display: 'inline-block' }}>
        <Typography sx={{ fontSize: 14 }} >
          <b>{review.user}</b>
        </Typography>
        <Box sx={{ fontSize: '11px', color: '#666666' }} >
          {moment(review.postedOn, 'YYYYMMDDTHHmmss').format('MMMM YYYY')}
        </Box>
      </Box>
      <Rating name="read-only" value={review.rating.overall} precision={0.5} readOnly sx={{ float: 'right', top: '20px', right: '20px' }} />
      <Typography sx={{
        fontSize: 14,
        lineHeight: 1.3,
        paddingTop: 2,
        paddingRight: 2,
        ...additionalStyle
      }}>
        {review.comment}
      </Typography>
    </Box>
  );
};

ReviewItem.propTypes = {
  review: PropTypes.object,
  additionalStyle: PropTypes.object
}

export default ReviewItem;
