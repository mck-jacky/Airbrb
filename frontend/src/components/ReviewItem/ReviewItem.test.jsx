import React from 'react';
import { render, screen } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow'
import ReviewItem from './ReviewItem'

const reviewObj = {
  user: 'Tom',
  postedOn: '20221113T112233',
  rating: 3.5,
  comment: 'This is a good comment'
}

describe('ReviewItem', () => {
  it('is in the document', () => {
    render(<ReviewItem review={reviewObj} additionalStyle={{}} />)
    const component = screen.getByTestId('review-item')
    // screen.debug()
    expect(component).toBeInTheDocument();
  })

  it('has svg', () => {
    render(<ReviewItem review={reviewObj} additionalStyle={{}} />)
    const component = screen.getByTestId('review-item')
    expect(component).toContainHTML('<svg');
  })

  it('has review user name', () => {
    render(<ReviewItem review={reviewObj} additionalStyle={{}} />)
    const component = screen.getByTestId('review-item')
    expect(component).toContainHTML('<b>Tom</b>');
  })

  it('has rating', () => {
    render(<ReviewItem review={reviewObj} additionalStyle={{}} />)
    const component = screen.getByTestId('review-item')
    expect(component).toContainHTML('3.5');
  })

  it('has comment date', () => {
    render(<ReviewItem review={reviewObj} additionalStyle={{}} />)
    const component = screen.getByTestId('review-item')
    expect(component).toContainHTML('November 2022');
  })

  it('has comment', () => {
    render(<ReviewItem review={reviewObj} additionalStyle={{}} />)
    const component = screen.getByTestId('review-item')
    expect(component).toContainHTML('This is a good comment');
  })

  it('has to match the snapshot', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<ReviewItem review={reviewObj} additionalStyle={{}} />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
