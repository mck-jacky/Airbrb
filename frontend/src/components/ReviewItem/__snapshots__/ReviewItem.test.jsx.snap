// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewItem has to match the snapshot 1`] = `
<ForwardRef(Box)
  data-testid="review-item"
>
  <Memo
    sx={
      Object {
        "fontSize": 44,
        "left": -5,
        "position": "relative",
        "top": 8,
      }
    }
  />
  <ForwardRef(Box)
    sx={
      Object {
        "display": "inline-block",
      }
    }
  >
    <ForwardRef(Typography)
      sx={
        Object {
          "fontSize": 14,
        }
      }
    >
      <b>
        Tom
      </b>
    </ForwardRef(Typography)>
    <ForwardRef(Box)
      sx={
        Object {
          "color": "#666666",
          "fontSize": "11px",
        }
      }
    >
      November 2022
    </ForwardRef(Box)>
  </ForwardRef(Box)>
  <ForwardRef(Rating)
    name="read-only"
    precision={0.5}
    readOnly={true}
    sx={
      Object {
        "float": "right",
        "right": "20px",
        "top": "20px",
      }
    }
  />
  <ForwardRef(Typography)
    sx={
      Object {
        "fontSize": 14,
        "lineHeight": 1.3,
        "paddingRight": 2,
        "paddingTop": 2,
      }
    }
  >
    This is a good comment
  </ForwardRef(Typography)>
</ForwardRef(Box)>
`;
